core/runtime/src/main/java/io/quarkiverse/langchain4j/RegisterAiService.java

```diff
@@ -178,6 +191,18 @@ public RetrievalAugmentor get() {
         }
     }
 
+    /**
+     * Marker that is used to tell Quarkus to use the {@link ContentRetriever} that the user has configured as a CDI bean.
+     * If no such bean exists, then no content retriever will be used.
+     */
+    final class BeanIfExistsContentRetrieverSupplier implements Supplier<ContentRetriever> {
```

Comment:
There should probably also be  a `NoContentRetrieverSupplier`  class, just like we have `NoRetrievalAugmentorSupplier`, which will mean "don't attempt to auto-detect a content retriever supplier"